File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraPivotGrid;
using System.Collections;

namespace Q181341 {
	public partial class Form1 : Form {
		IList dataSource;
		PropertyDescriptor orderPropDesc;
		PivotGridField field;

		public Form1() {
			InitializeComponent();

			field = pivotGridControl1.Fields.Add("Desc", PivotArea.ColumnArea);		
			pivotGridControl1.CustomFieldSort += new PivotGridCustomFieldSortEventHandler(pivotGridControl1_CustomFieldSort);

			dataSource = CreateData().DefaultView;
			orderPropDesc = ((ITypedList)dataSource).GetItemProperties(null)["Order"];

			pivotGridControl1.DataSource = dataSource;
		}

		void pivotGridControl1_CustomFieldSort(object sender, PivotGridCustomFieldSortEventArgs e) {
			if(e.Field.FieldName != "Desc") return;
			int order1 = Convert.ToInt32(orderPropDesc.GetValue(dataSource[e.ListSourceRowIndex1])),
				order2 = Convert.ToInt32(orderPropDesc.GetValue(dataSource[e.ListSourceRowIndex2]));
			e.Result = Comparer<int>.Default.Compare(order1, order2);
			e.Handled = true;
		}

		DataTable CreateData() {
			DataTable res = new DataTable();
			res.Columns.Add("Order", typeof(int));
			res.Columns.Add("Desc", typeof(string));
			res.Rows.Add(1, "January");
			res.Rows.Add(2, "February");
			res.Rows.Add(3, "March");
			res.Rows.Add(4, "April");
			return res;
		}

		private void checkBox1_CheckedChanged(object sender, EventArgs e) {
			field.SortMode = checkBox1.Checked ? PivotSortMode.Custom : PivotSortMode.Default;
		}
	}
}

File: VB/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraPivotGrid
Imports System.Collections

Namespace Q181341

    Public Partial Class Form1
        Inherits Form

        Private fieldMonth As PivotGridField

        Public Sub New()
            InitializeComponent()
            fieldMonth = pivotGridControl1.Fields.Add("Month", PivotArea.ColumnArea)
            AddHandler pivotGridControl1.CustomFieldSort, New PivotGridCustomFieldSortEventHandler(AddressOf pivotGridControl1_CustomFieldSort)
            pivotGridControl1.DataSource = CreateData()
        End Sub

        Private Sub pivotGridControl1_CustomFieldSort(ByVal sender As Object, ByVal e As PivotGridCustomFieldSortEventArgs)
            If Equals(e.Field.FieldName, "Month") Then
                Dim orderValue1 As Object = e.GetListSourceColumnValue(e.ListSourceRowIndex1, "Order"), orderValue2 As Object = e.GetListSourceColumnValue(e.ListSourceRowIndex2, "Order")
                e.Result = Comparer.Default.Compare(orderValue1, orderValue2)
                e.Handled = True
            End If
        End Sub

        Private Function CreateData() As DataTable
            Dim res As DataTable = New DataTable()
            res.Columns.Add("Order", GetType(Integer))
            res.Columns.Add("Month", GetType(String))
            res.Rows.Add(1, "January")
            res.Rows.Add(2, "February")
            res.Rows.Add(3, "March")
            res.Rows.Add(4, "April")
            Return res
        End Function

        Private Sub checkBox1_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs)
            fieldMonth.SortMode = If(checkBox1.Checked, PivotSortMode.Custom, PivotSortMode.Default)
        End Sub
    End Class
End Namespace
